src/components/EditableFields/AutoSave.tsx

```diff
@@ -16,6 +17,43 @@ interface AutoSaveProps {
   children?: React.ReactNode
 }
 
+const excludeSavedAmendments = (
+  amendmentFields: AmendmentKeys[],
+  amendments: Amendments,
+  savedAmendments: Amendments
+) => {
+  const map = new Map()
+
+  amendmentFields?.forEach((amendmentField) => {
```

Comment:
This will remove one nesting level:

```suggestion
  amendmentFields?.filter(amendmentField => Array.isArray(amendments[amendmentField])).forEach((amendmentField) => {
```